packages/lib-classifier/src/store/Subjects.js

```diff
@@ -0,0 +1,51 @@
+import { addDisposer, getRoot, onPatch, flow, types } from 'mobx-state-tree'
+import Subject from './Subject'
+
+const Subjects = types
+  .model('Subjects', {
+    queue: types.optional(types.array(Subject), [])
+  })
+
+  .views(self => ({
+    get current () {
+      return self.queue.length ? self.queue[0] : null
+    }
+  }))
+
+  .actions(self => ({
+    addCurrentWorkflowListener () {
+      const { workflows } = getRoot(self)
+      const populateQueueDisposer = onPatch(workflows, call => {
+        if (call.path === '/current') {
+          return self.populate()
+        }
+      })
+      addDisposer(self, populateQueueDisposer)
+    },
+
+    advance () {
+      self.queue.shift()
+      if (self.queue.length < 3) {
+        self.populate()
+      }
+    },
+
+    afterAttach () {
+      self.addCurrentWorkflowListener()
+    },
+
+    populate: flow(function * populate () {
+      const { client, workflows } = getRoot(self)
+      const workflowId = workflows.current.id
```

Comment:
Is it possible for workflows.current to be undefined? If project does not have any linked workflows for some reason or if the linked workflows are inactive? We might want an existential check here.

Reply:
It _shouldn't_ be, but I've put in a check on this method. I did think about putting it in the listener, but as it's currently a public method here is probably the right place.